source/functions/BEMIO/formatPlot.m

```diff
@@ -45,47 +45,52 @@ function formatPlot(fig,titleString,subtitleStrings,xString,yString,X,Y,legendSt
 
 legendStrings1D = legendStrings(:);
 legendStrings1D = legendStrings1D(~cellfun(@isempty, legendStrings1D));
-
-% Subplot formatting
-nPlots = length(subtitleStrings);
-if nPlots < 4
-    isp1 = 1;
-    isp2 = nPlots;
-else
-    isp1 = 2;
-    isp2 = ceil(nPlots/2);
-end
+legendStrings1D = strrep(legendStrings1D,'_','\_');
 
 % Figures
 nHydro = length(fieldnames(X));
 nBodiesTotal = 0;
 for iHydro = 1:nHydro
     tmp1 = strcat('X',num2str(iHydro));
     tmp2 = strcat('Y',num2str(iHydro));
-    [~,nBodies,~] = size(Y.(tmp2));
+    if isstruct(Y.(tmp2))
+        [~,nBodies,~] = size(Y.(tmp2).d1);
+        nDirections = length(fieldnames(Y.(tmp2)));
+    else
+        [~,nBodies,~] = size(Y.(tmp2));
+        nDirections = 0;
+    end
     nBodiesTotal = nBodiesTotal + nBodies;
 
-    % Surge
-%     subplot('Position',[0.0731 0.3645 0.2521 0.4720])
-    for isp = 1:nPlots
-        subplot(isp1,isp2,isp);
+    for iDofs = 1:size(plotDofs,1)
+        
+        % iterates through each degree of freedom
+        subplot('Position',[.2/(size(plotDofs,1))+((.975/size(plotDofs,1))*(iDofs-1)) 0.3645 0.756/size(plotDofs,1) 0.4720])
         hold('on');
         box('on');
         for iBody = 1:nBodies
-            plot(X.(tmp1),squeeze(Y.(tmp2)(isp,iBody,:)),'LineWidth',1)  
+            if nDirections>0
+                for iDir = 1:nDirections
+                    tmp3 = strcat('d',num2str(iDir));
+                    plot(X.(tmp1),squeeze(Y.(tmp2).(tmp3)(iDofs,iBody,:)),'LineWidth',1)  
+                end
+            else
+                plot(X.(tmp1),squeeze(Y.(tmp2)(iDofs,iBody,:)),'LineWidth',1)
+            end
         end
```

Comment:
Is there an issue that this resolves?

Reply:
Not really an issue, but I think the expected behavior is to have ``hold off`` once we're done editing a figure. If a user tried to plot something again, it will automatically add onto the new plot instead of creating a new one.